src/joypad.c

```diff
@@ -725,12 +746,41 @@ void joypad_poll(void)
     }
 
     if (check_origins) joypad_gcn_origin_check_async();
+
+    if (sys_bbplayer())
+    {
+        joypad_device_cold_t swap = joypad_devices_cold[JOYPAD_PORT_1];
+        // Respect the BBPlayer "Hack Flags" register to swap controller ports
+        switch( BB_HACK_FLAGS_SWAP_PORT1 )
+        {
+            case 1:
+                // Swap port 1 and 2
+                joypad_devices_cold[JOYPAD_PORT_1] = joypad_devices_cold[JOYPAD_PORT_2];
+                joypad_devices_cold[JOYPAD_PORT_2] = swap;
+                break;
+            case 2:
+                // Swap port 1 and 3
+                joypad_devices_cold[JOYPAD_PORT_1] = joypad_devices_cold[JOYPAD_PORT_3];
+                joypad_devices_cold[JOYPAD_PORT_3] = swap;
+                break;
+            case 3:
+                // Swap port 1 and 4
+                joypad_devices_cold[JOYPAD_PORT_1] = joypad_devices_cold[JOYPAD_PORT_4];
+                joypad_devices_cold[JOYPAD_PORT_4] = swap;
+                break;
+        }
+    }
 }
 
 bool joypad_is_connected(joypad_port_t port)
 {
     ASSERT_JOYPAD_INITIALIZED();
     ASSERT_JOYPAD_PORT_VALID(port);
+
+    // Respect the BBPlayer "Hack Flags" register to swap controller ports.
+    // The device state is swapped during poll, but identifiers need to be swapped on read.
+    port = bb_hack_flags_swap_port(port);
+
```

Comment:
Why do `joypad_is_connected` and `joypad_get_identifier` use the `hot` structure rather than the `cold`? I noticed since they are the only two functions that require an explicit fix for the BB swap, but I was wondering what is the rationale.